src/input/reader_to_storage.cc

```diff
@@ -347,17 +347,73 @@ StorageBase * ReaderToStorage::make_storage(PathBase &p, const Type::Array *arra
                     << EI_ErrorAddress(p.as_string()) << EI_InputType(array->desc()) );
         }
     } else {
-        // try automatic conversion to array with one element
-        if ( array->match_size( 1 ) ) {
-            StorageArray *storage_array = new StorageArray(1);
-            const Type::TypeBase &sub_type = array->get_sub_type();
-            storage_array->new_item(0, make_storage(p, &sub_type) );
-
-            return storage_array;
-        } else {
-            THROW( ExcInputError() << EI_Specification("Automatic conversion to array not allowed. The value should be '" + p.get_node_type(ValueTypes::array_type) + "', but we found: ")
-                    << EI_ErrorAddress(p.as_string()) << EI_JSON_Type( p.get_node_type(p.get_node_type_index()) ) << EI_InputType(array->desc()) );
+    	// if transposition is carried, only conversion to array with one element is allowed
+    	if (try_transpose_read_) {
+			// try automatic conversion to array with one element
+			if ( array->match_size( 1 ) ) {
+				StorageArray *storage_array = new StorageArray(1);
+				const Type::TypeBase &sub_type = array->get_sub_type();
+				storage_array->new_item(0, make_storage(p, &sub_type) );
+
+				return storage_array;
+			} else {
+				THROW( ExcInputError() << EI_Specification("Automatic conversion to array not allowed. The value should be '" + p.get_node_type(ValueTypes::array_type) + "', but we found: ")
+						<< EI_ErrorAddress(p.as_string()) << EI_JSON_Type( p.get_node_type(p.get_node_type_index()) ) << EI_InputType(array->desc()) );
+			}
         }
+
+    	// set variables managed transposition
+        try_transpose_read_ = true;
+    	transpose_index_ = 0;
+    	transpose_array_sizes_.clear();
+
+    	const Type::TypeBase &sub_type = array->get_sub_type();
+    	StorageBase *tmp_storage = make_storage(p, &sub_type);
```

Comment:
Here we should catch thrown exceptions, form their messages and throw new exception with message:
Failed transpose auto-conversion of the intput EI_InputType to the type EI_RequiredType. 
Address: EI_Address
Transpose filed due to: <original message>
Address: EI_Address